src/test/java/htsjdk/samtools/util/IntervalTest.java

```diff
@@ -19,4 +23,86 @@ public void testGetStrand(boolean isNegativeStrand){
         Assert.assertNotEquals(isNegativeStrand, interval.isPositiveStrand());
         Assert.assertEquals(isNegativeStrand, interval.getStrand() == Strand.NEGATIVE);
     }
+
+    @DataProvider
+    public Object[][] getIntervalsForIntersectionAndAbutment(){
+        final String chr1 = "chr1";
+        final String chr2 = "chr2";
+
+        final Interval chr1_10_20 = new Interval(chr1, 10, 20);
+        final Interval chr1_10_21 = new Interval(chr1, 10, 21);
+        final Interval chr1_10_22 = new Interval(chr1, 10, 22);
+        final Interval chr1_10_23 = new Interval(chr1, 10, 23);
+        final Interval chr1_20_30 = new Interval(chr1, 20, 30);
+        final Interval chr1_21_30 = new Interval(chr1, 21, 30);
+        final Interval chr1_22_30 = new Interval(chr1, 22, 30);
+        final Interval chr1_23_30 = new Interval(chr1, 23, 30);
+        final Interval chr1_24_30 = new Interval(chr1, 24, 30);
+        final Interval zeroLength_chr1_22_21 = new Interval(chr1, 22, 21);
+        final Interval chr2_10_20 = new Interval(chr2, 10, 20);
+        final Interval chr2_21_30 = new Interval(chr2, 21,  30);
+
+        /*
+
+           10-20  <---|
+           10-21  <--------|
+           10-22  <-------------|
+           10-23  <------------------|
+           22-21                |
+           20-30 |----------------------->
+           21-30      |------------------>
+           22-30           |------------->
+           23-30                |-------->
+           24-30                     |--->
+             ... | 20 | 21 | 22 | 23 | 24
+         */
+
+        final Object[][] cases = new Object[][]{
+                //interval1, interval2, intersects, abuts
+                {chr1_10_20, chr1_10_20, true, false},
+                {chr1_10_20, chr1_20_30, true, false},
+                {chr1_10_20, chr1_21_30, false, true},
+                {chr1_10_20, chr1_22_30, false, false},
+                {chr1_10_20, zeroLength_chr1_22_21, false, false},
+                {chr1_10_21, zeroLength_chr1_22_21, false, false},
+                {chr1_10_22, zeroLength_chr1_22_21, false, true},
+                {chr1_10_23, zeroLength_chr1_22_21, true, false},
+                {chr1_20_30, zeroLength_chr1_22_21, true, false},
+                {chr1_21_30, zeroLength_chr1_22_21, true, false},
+                {chr1_22_30, zeroLength_chr1_22_21, true, false},
+                {chr1_23_30, zeroLength_chr1_22_21, false, true},
+                {chr1_24_30, zeroLength_chr1_22_21, false, false},
+                {zeroLength_chr1_22_21, zeroLength_chr1_22_21, true, false},
```

Comment:
hmmm. to answer his question we need a definition of "intersects" and "abuts"

Intersects: There exists at least one base in the reference that is present in both intervals.
Abuts: Two intervals that do not intersect (c.f.) , however, the union of bases they cover can be written as one interval. 

According to this definition, two empty intervals in the same position do not intersect, but they do abut.

Reply:
we have both intersects and overlaps? huh?

Reply:
it's the same function, implemented twice!

Reply:
We have more in other places I'm sure.   Because of  fear or weird details like this we never merged the implementations of a lot of these.